doc/graphviz/cache_fetch.dot

```diff
@@ -43,9 +43,27 @@ digraph cache_fetch {
 		]
 		v_b_f:error:s -> v_b_e
 		v_b_f:fetch:s -> v_b_hdrs [style=bold]
+		RETRY_FETCH [shape=plaintext]
+		RETRY_FETCH -> v_b_hdrs [style=bold]
 		v_b_hdrs [ label="send bereq,\nread beresp (headers)"]
-		v_b_hdrs -> v_b_r [style=bold]
+		v_b_hdrs -> v_b_r [label="\"no stale object\""] [style=bold]
 		v_b_hdrs -> v_b_e
+		v_b_hdrs -> v_b_refresh [label="\"stale object\""] [style=bold]
+
+		v_b_refresh [
+			    shape=record
+			    label="{vbf_stp_startfetch:|{vcl_backend_refresh\{\}|{bereq.*|beresp.*}}|{<error>error|fail|{retry|{<max>max?|<retry>ok?}}|{retry(fetch)|{<rf_max>max?|<rf_retry>ok?}}|abandon|<merge>merge|<stale>obj_stale|<beresp>beresp}}"
+		]
+		v_b_refresh:error:s -> v_b_e
+		v_b_refresh:retry -> v_b_r_retry [color=purple]
+		v_b_refresh:rf_retry -> v_b_r_retry_fetch [color=purple]
```

Comment:
Maybe don't share the `vcl_backend_response` retry transition to have less arrows crossing others? (Just like `vcl_backend_fetch` has its dedicated retry transition.)

I'm also wondering, is there a specific reason not to allow `return (retry(fetch))` from `vcl_backend_response`?

Reply:
I meant to also include `vcl_backend_error` too.

I also think we should be fine with a single "retry" box, with two distinct arrows towards "RETRY" and "RETRY_FETCH". Especially since there are two max/ok sub "boxes".

Reply:
I guess we can generalize it later if there is an actual case for it.

Consider this pull request approved on my end, but please also consider my last review items (including the cosmetic side of this one).